src/libraries/System.Security.Principal.Windows/src/System/Security/Principal/IRCollection.cs

```diff
@@ -129,25 +129,20 @@ public IEnumerator<IdentityReference> GetEnumerator()
         {
             get
             {
-                return _Identities[index];
+                return _identities[index];
             }
 
             set
             {
-                if (value == null)
-                {
-                    throw new ArgumentNullException(nameof(value));
-                }
-
-                _Identities[index] = value;
+                _identities[index] = value ?? throw new ArgumentNullException(nameof(value));
```

Comment:
Personally, I hate this syntax, especially for argument validation.  (Mainly because it doesn't compose well, because it leaks half-done work when used for field assignments).

Since there's not a second statement here, and there's unlikely to ever be one, I don't have a technical argument.  So I'll just leave it at babbling.

Reply:
It was an autofix suggestion so I have no attachment to it. Informed babbling is reason enough for me, reverted.